rust/tests/loadtest/src/config.rs

```diff
@@ -0,0 +1,904 @@
+    pub step: Option<u64>,
+}
+
+impl Range {
+    /// Validate the range configuration.
+    pub fn validate(&self, field_name: &str, section: &str) -> Result<(), ConfigError> {
+        if self.min > self.max {
+            return Err(ConfigError::InvalidRange {
+                section: section.to_string(),
+                field: field_name.to_string(),
+                min: self.min,
+                max: self.max,
+            });
+        }
+
+        if let Some(step) = self.step
+            && step == 0
+        {
+            return Err(ConfigError::ZeroStep {
+                section: section.to_string(),
+                field: field_name.to_string(),
+            });
+        }
+
+        Ok(())
+    }
+
+    /// Pick a random value from the range.
+    pub fn pick<R: Rng>(&self, rng: &mut R) -> u64 {
+        match self.step {
+            Some(step) => {
+                let num_steps = (self.max - self.min) / step;
+                let chosen_step = rng.gen_range(0..=num_steps);
+                self.min + chosen_step * step
+            }
+            None => rng.gen_range(self.min..=self.max),
+        }
+    }
+}
+
+/// HTTP load test configuration.
+#[derive(Debug, Deserialize)]
+pub struct HttpConfig {
+    /// List of target URLs to choose from.
+    pub addresses: Vec<String>,
+    /// HTTP versions to choose from (1 or 2).
+    pub http_version: Vec<u8>,
+    /// Number of concurrent users.
+    pub users: Range,
+    /// Test duration in seconds.
+    pub run_time_secs: Range,
+}
+
+impl HttpConfig {
+    fn validate(&self) -> Result<(), ConfigError> {
+        if self.addresses.is_empty() {
+            return Err(ConfigError::EmptyAddresses {
+                section: "http".to_string(),
+            });
+        }
+
+        // Validate all URLs can be parsed
+        for addr in &self.addresses {
+            Url::parse(addr).map_err(|e| ConfigError::InvalidUrl {
+                section: "http".to_string(),
+                url: addr.clone(),
+                error: e.to_string(),
+            })?;
+        }
+
+        if self.http_version.is_empty() {
+            return Err(ConfigError::EmptyHttpVersions);
+        }
+        for v in &self.http_version {
+            if *v != 1 && *v != 2 {
+                return Err(ConfigError::InvalidHttpVersion { version: *v });
+            }
+        }
+
+        self.users.validate("users", "http")?;
+        self.run_time_secs.validate("run_time_secs", "http")?;
+
+        Ok(())
+    }
+}
+
+/// TCP load test configuration.
+#[derive(Debug, Deserialize)]
+pub struct TcpConfig {
+    /// List of target addresses (host:port) to choose from.
+    pub addresses: Vec<String>,
+    /// Number of concurrent connections.
+    pub concurrent: Range,
+    /// How long to hold connections in seconds.
+    pub duration_secs: Range,
+    /// Connection timeout in seconds.
+    pub timeout_secs: Range,
+    /// Enable echo mode for payload verification.
+    pub echo_mode: bool,
```

Comment:
I don't think we need a non-echo mode, we should always send messages. Also keeps the code simpler here.